arbor/backends/gpu/shared_state.hpp

```diff
@@ -34,7 +34,7 @@ struct ion_state {
     array weight_Xi_;   // (1) concentration weight internal
     array weight_Xo_;   // (1) concentration weight external
 
-    int charge;    // charge of ionic species
+    array charge;       // charge of ionic species (global, length 1)
```

Comment:
Why do we need this to be an array?

Reply:
Mainly because `array` handles automatically all the GPU memory management for us. A pointer would suffice, but would complicate the allocation and destruction somewhat.